packages/mask/src/components/shared/ApplicationBoard.tsx

```diff
@@ -86,350 +20,23 @@ const useStyles = makeStyles()((theme) => {
-                    SUPPORTED_CHAIN_ID_LIST,
-                ),
-            undefined,
-            true,
-        ),
-        createEntry(
-            'Alternative',
-            new URL('./assets/more.png', import.meta.url).toString(),
-            () =>
-                openSecondEntryDir(
-                    'Alternative',
-                    [
-                        createEntry(
-                            'PoolTogether',
-                            new URL('./assets/pool_together.png', import.meta.url).toString(),
-                            () => {},
-                        ),
-                    ],
-                    SUPPORTED_CHAIN_ID_LIST,
-                ),
-            undefined,
-            true,
-        ),
-        createEntry(
-            'FindTruman',
-            new URL('./assets/findtruman.png', import.meta.url).toString(),
-            onFindTrumanDialogOpen,
-            [ChainId.Mainnet],
-            isNotEvm,
-            true,
-        ),
-    ]
-
+    const snsAdaptorPlugins = useActivatedPluginsSNSAdaptor('any')
     return (
         <>
-            {secondEntryChainTabs?.length ? (
-                <div className={classes.abstractTabWrapper}>
-                    <NetworkTab
-                        chainId={chainId}
-                        setChainId={setChainId}
-                        classes={classes}
-                        chains={secondEntryChainTabs}
-                    />
-                </div>
-            ) : null}
             <section className={classes.applicationWrapper}>
-                {(secondEntries ?? firstLevelEntries).map(
-                    ({ title, img, onClick, supportedChains, hidden, walletRequired }, i) =>
-                        (!supportedChains || supportedChains?.includes(chainId)) && !hidden ? (
-                            <div
-                                className={classNames(
-                                    classes.applicationBox,
-                                    walletRequired && !selectedWallet ? classes.disabled : '',
-                                )}
-                                onClick={onClick}
-                                key={i}>
-                                <img src={img} className={classes.applicationImg} />
-                                <Typography className={classes.title} color="textPrimary">
-                                    {title}
-                                </Typography>
-                            </div>
-                        ) : null,
-                )}
+                {snsAdaptorPlugins
+                    .reduce<Plugin.SNSAdaptor.ApplicationEntry[]>((acc, cur) => {
+                        if (!cur.ApplicationEntries) return acc
+                        return acc.concat(cur.ApplicationEntries ?? [])
+                    }, [])
+                    .sort((a, b) => a.defaultSortingPriority - b.defaultSortingPriority)
+                    .map((x, i) => x.RenderEntryComponent(i))}
```

Comment:
Please do not let the plugin handle the `key` property and do not call components as function (this will fragile when those components use hooks). Do this:

```js
.map(X => <React.Fragment key={X.ID}><X.RenderEntryComponent /></React.Fragment>)
```

Reply:
thx done.